src/tracer.js

```diff
@@ -297,12 +305,7 @@ export default class Tracer {
             throw new Error(`Unsupported format: ${format}`);
         }
 
-        let spanContext = extractor.extract(carrier);
-
-        if (spanContext) {
-            spanContext.finalizeSampling();
-        }
-        return spanContext;
+        return extractor.extract(carrier);
```

Comment:
No, none of the tests break because of this. In this implementation,  finalizing a span removes the deferred sampling flag, so finalizing a span before we apply a sampling decision is problematic.